### PR TITLE
Add test for 'ipfs config replace'

### PR DESCRIPTION
**Pull request by Kubuxu**

### CODE REVIEW

File: test/sharness/t0021-config.sh [hunk 1]
```diff
@@ -81,8 +94,6 @@ test_config_cmd() {
        test_cmp ident_exp ident_out
   '
 
-  # SECURITY
-  # Those tests are here to prevent exposing the PrivKey on the network
```
Comment (whyrusleeping) [inline test/sharness/t0021-config.sh]: whyd you remove this comment?

Comment (whyrusleeping) [inline test/sharness/t0021-config.sh]: oh nvm, i see that you moved it up

**Comment by whyrusleeping** [comment]
Can you maybe break that test up a little bit? maybe into a setup step, the command execution, and the output check?

**Comment by whyrusleeping** [comment]
@Kubuxu any update here?

**Comment by Kubuxu** [comment]
How does it look now?

**Comment by whyrusleeping** [comment]
Much better, except you've angered the mighty travis CI.

**Comment by Kubuxu** [comment]
@whyrusleeping OSX/BSD sed is behaving quite strangely with `-i` flag and it looks like it will not support search and execute. Can I add gnu-sed as a requirement on OSX/BSD based systems?

//cc @chriscool

**Comment by chriscool** [comment]
@Kubuxu I think it is better to just avoid using the -i option when using sed.
You can instead redirect into new files for example like:

```
	sed -e /PrivKey/d -e s/10GB/11GB/ newconfig.orig.json >newconfig.step.json &&
	sed -e '"'"'/PeerID/ { s/,$// } '"'"' newconfig.step.json >newconfig.json
```

**Comment by chriscool** [comment]
It looks like there are still sed related problems:

```
expecting success: 

    cp "$IPFS_PATH/config" newconfig.json &&
	</dev/null sed -e /PrivKey/d -e s/10GB/11GB/ -i"~"  newconfig.json &&
	</dev/null sed -e '/PeerID/ { s/,$// } ' -i"~" newconfig.json

sed: 1: "/PeerID/ { s/,$// } 
": bad flag in substitute command: '}'
not ok 49 - setup for config replace test
```

**Comment by Kubuxu** [comment]
It works, Shanress on OSX passed.

**Comment by whyrusleeping** [comment]
LGTM, thanks @Kubuxu and @chriscool :)

**Pull request merged**